### DISCUSSION TITLE
[Feature]: Goal to create proto archives

### DISCUSSION THREAD

**Discussion by ascopes** [ascopes/protobuf-maven-plugin#462] [Ideas]
A common pattern that I have seen is the use of `zip` archives containing protobuf sources
that are then added as a dependency by other projects.

Currently, if a Maven project wishes to create one of these ZIPs, it is a somewhat manual 
process that involves setting up the Maven Archiver plugin in a specific way.

I think we could make that easier for users by potentially having a goal in this project that
concerns itself with generating these kinds of archives automatically.

One idea could be to introduce a new project packaging type that limits the Maven phases that get run so that only the required steps get called, e.g:

```
<project>
  ...
  <packaging>protobuf-zip</packaging>
  ...
</project>
```

This would then be able to register the ZIP as the project artifact for deployment to a Maven Central registry or similar.

**Comment by cdprete** [comment]
It would be nice indeed, but then you should somehow offer the same flexibility the archiver plugin offers today I would say.

What about its delivery then? Would the generated archive be attached to the project as additional artifact or not?
If yes, can it be opted out if needed? If no, why not?

I think it's quite a complex topic for, maybe, little added value since you may do it cleanly with Maven multi-modules.
Just have a submodule that contains the .proto as sources and ships them as zip or as sources jar.
Then the project which needs to generate the classes from the .proto can decide to have such zip artifact as dependency and instruct Maven to unzip it or to just use the sources jar as dependency directly.

**Reply by ascopes** [reply]
This raises a good point regarding the value versus compexity I think. This is my concern as well.

> Would it be attached

Yep, that would be the idea hopefully.

> Have the zip archive ... maven to unzip it

Right now, we treat WAR, EAR, JAR, and ZIP archives as one and the same, so will automatically interrogate them for sources, so this side of it is already fairly fluent (hopefully). It should be smart enough to be given these archives as regular project dependencies.

**Reply by cdprete** [reply]
Just my 2 cents 😂
